deployments/pulse3d/services/pulse3d/tests/test_main.py

```diff
@@ -769,14 +831,14 @@ def test_jobs__post__omits_analysis_params_not_supported_by_the_selected_pulse3d
     pulse3d_semver = VersionInfo.parse(version)
     if pulse3d_semver >= "0.25.0":
         expected_analysis_param_keys.append("max_y")
-    if pulse3d_semver >= "0.25.2":
-        expected_analysis_param_keys.append("peaks_valleys")
     if pulse3d_semver >= "0.25.4":
         expected_analysis_param_keys.append("normalize_y_axis")
     if pulse3d_semver >= "0.26.0":
         expected_analysis_param_keys.append("stiffness_factor")
     if pulse3d_semver >= "0.27.4":
         expected_analysis_param_keys.append("inverted_post_magnet_wells")
+    if "0.25.2" <= pulse3d_semver < "0.28.0":
```

Comment:
Peaks and valleys will not be supported in versions later than 0.28.0?

Reply:
On line 144 in `/pulse3d/src/main.py`, this is where peaks and valleys get handled with versions greater than 0.28.0. This line `"0.25.2" <= pulse3d_semver < "0.28.0"` allows compatibility with older pulse3d versions so that users can still use it with recordings analyzed with older versions.